src/game/BattleGround/BattleGroundWS.cpp

```diff
@@ -133,13 +131,13 @@ void BattleGroundWS::RespawnDroppedFlag(Team team)
     if (GetStatus() != STATUS_IN_PROGRESS)
         return;
 
+    PvpTeamIndex teamIdx = GetTeamIndexByTeamId(team);
     RespawnFlag(team, false);
-    if (team == ALLIANCE)
-        SendMessageToAll(LANG_BG_WS_ALLIANCE_FLAG_RESPAWNED, CHAT_MSG_BG_SYSTEM_NEUTRAL);
-    else
-        SendMessageToAll(LANG_BG_WS_HORDE_FLAG_RESPAWNED, CHAT_MSG_BG_SYSTEM_NEUTRAL);
 
-    PlaySoundToAll(BG_WS_SOUND_FLAGS_RESPAWNED);
+    SendMessageToAll(wsMessageIds[teamIdx][BG_WS_FLAG_ACTION_CAPTURED],
+        wsChatMessageTypes[teamIdx][BG_WS_FLAG_ACTION_CAPTURED]);
```

Comment:
I’ll take this as an example for all your similar “optimizations”.. all you are doing is adding array lookups since the branch (maybe even two branches) still exists in GetTeamIndexByTeamId. Yes there is less code duplication but at the price of performance and i don’t think BG code needs to be touched often if it’s working

Reply:
Well to be fair, there was game breaking problems that needed to be addressed.

Reply:
No one said we can't touch the code if there are issues (game breaking or otherwise). I think he added "if it's working" as a condition for a reason :smile:

Reply:
Which it wasn't & even this rework might have more issues. The point is that cleaner code makes for easier debugging, that was my whole reason to rework this class.

Reply:
I probably took it to literally :P I have a habit of doing that.